rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt

```diff
@@ -1219,6 +1218,97 @@ data class DisplayStmt(val factor1: Expression?, val response: Expression?, over
     }
 }
 
+@Serializable
+data class DOWxxStmt(
+    val comparison: ComparisonOperator,
+    val factor1: Expression,
+    val factor2: Expression,
+    override val body: List<Statement>,
+    override val position: Position? = null
+) : Statement(position), CompositeStatement {
+    fun InterpreterCore.compare(comparison: ComparisonOperator, factor1: Expression, factor2: Expression, interpreter: InterpreterCore): Boolean {
```

Comment:
You have managed a specific  comparision logic just for `DOWxx` statement, and that it means that the result of `value1 == value2 or value1 >= value2... `could be different depending on statement.
Since that you are comparing two instances of Value you should manage this logic in the `operator fun compareTo`
in the concrete implementation of Value, which is always the right place to implement this kind of logic.

Reply:
So, if I understand correctly, I have to move the implementation about `.compare` into `Value`